pumpkin-world/src/world_gen/implementations/superflat.rs

```diff
@@ -0,0 +1,47 @@
+use crate::{
+    biome::Biome,
+    block::BlockId,
+    coordinates::{BlockCoordinates, XZBlockCoordinates},
+    world_gen::{
+        generator::{BiomeGenerator, GeneratorInit, TerrainGenerator},
+        generic_generator::GenericGenerator,
+        Seed,
+    },
+};
+
+#[allow(dead_code)]
+pub type SuperflatGenerator = GenericGenerator<SuperflatBiomeGenerator, SuperflatTerrainGenerator>;
+
+pub(crate) struct SuperflatBiomeGenerator {}
+
+impl GeneratorInit for SuperflatBiomeGenerator {
+    fn new(_: Seed) -> Self {
+        Self {}
+    }
+}
+
+impl BiomeGenerator for SuperflatBiomeGenerator {
+    // TODO make generic over Biome and allow changing the Biome in the config.
+    fn generate_biome(&self, _: XZBlockCoordinates) -> Biome {
+        Biome::Plains
+    }
+}
+
+pub(crate) struct SuperflatTerrainGenerator {}
+
+impl GeneratorInit for SuperflatTerrainGenerator {
+    fn new(_: Seed) -> Self {
+        Self {}
+    }
+}
+
+impl TerrainGenerator for SuperflatTerrainGenerator {
+    // TODO allow specifying which blocks should be at which height in the config.
+    fn generate_block(&self, at: BlockCoordinates, _: Biome) -> BlockId {
+        if *at.y == 0 {
+            BlockId::from_id(9)
+        } else {
+            BlockId::AIR
+        }
+    }
+}
```

Comment:
Lets do it like vanilla does

First layer -> Grass
2 Blocks -> Dirt
Last layer -> Bedrock

Reply:
![image](https://github.com/user-attachments/assets/98cd6c89-ccfa-4b73-b409-68a5ce1c60b0)